subsys/debug/coredump/Kconfig

```diff
@@ -28,6 +28,12 @@ config DEBUG_COREDUMP_BACKEND_FLASH_PARTITION
 	  Core dump is saved to a flash partition with DTS alias
 	  "coredump-partition".
 
+config DEBUG_COREDUMP_BACKEND_MEM_WINDOW
+	bool "Use memory window 2 slot [1] space for coredump"
+	depends on !LOG
+	help
+	  Core dump is done via memory window slot[1].
```

Comment:
I think this needs some dependencies and additonal context explaining what the memory windows are in this case. I mean this refers to very specific implementation of memory windows on Intel audio DSPs. So I'd even rename this to DEBUG_COREDUMP_BACKEND_INTEL_ADSP (or something to that effect).